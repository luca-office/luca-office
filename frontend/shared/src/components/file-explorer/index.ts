export * from "./file-explorer"
export * from "./tree-node"
