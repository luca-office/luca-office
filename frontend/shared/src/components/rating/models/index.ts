export * from "./criterion-metadata"
export * from "./rating-coding-criterion"
export * from "./scenario-criterion-selection"
export * from "./table-entity"
