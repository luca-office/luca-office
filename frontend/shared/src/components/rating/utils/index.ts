export * from "./coding-criterion"
export * from "./coding-dimension"
export * from "./coding-item"
export * from "./question"
export * from "./score"
export * from "./scoring-type"
