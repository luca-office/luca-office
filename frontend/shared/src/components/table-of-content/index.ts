export * from "./table-of-contents"
export * from "./table-of-contents-entry/table-of-contents-entry"
