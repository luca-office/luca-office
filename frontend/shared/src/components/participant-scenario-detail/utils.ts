import {FreetextAnswerFragment} from "../../graphql/generated/FreetextAnswerFragment"
import {QuestionType} from "../../graphql/generated/globalTypes"
import {Questionnaire, RuntimeSurveyResults} from "../../models"
import {ParticipantQuestionAnswer, QuestionnaireParticipantResults} from "../questionnaire"

export const runtimeSurveyResultsToQuestionnaireParticipantResults = (
  invitationId: UUID,
  questionnaire: Questionnaire,
  runtimeSurveyResults: RuntimeSurveyResults,
  freeTextAnswers: FreetextAnswerFragment[]
): QuestionnaireParticipantResults => {
  const answers = runtimeSurveyResults.questionResults.reduce<Record<UUID, ParticipantQuestionAnswer>>(
    (answersAccumulator, result) => {
      const question = questionnaire.questions.find(question => question.id === result.questionId)
      const participantResult = result.participantResults.find(results => results.invitationId === invitationId)
      const freeTextAnswer = freeTextAnswers.find(
        answer => answer.questionId === result.questionId && answer.surveyInvitationId === invitationId
      )?.text

      switch (question?.questionType) {
        case QuestionType.SingleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.SingleChoice,
              questionId: question.id,
              selectedAnswerId: participantResult?.selectedAnswerIds[0] ?? "",
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.MultipleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.MultipleChoice,
              questionId: question.id,
              selectedAnswerIds: participantResult?.selectedAnswerIds ?? [],
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.FreeText: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.FreeText,
              questionId: question.id,
              freeTextAnswer: freeTextAnswer ?? ""
            }
          }
        }

        default:
          return answersAccumulator
      }
    },
    {}
  )

  return {questionnaireId: runtimeSurveyResults.questionnaireId, answers}
}
