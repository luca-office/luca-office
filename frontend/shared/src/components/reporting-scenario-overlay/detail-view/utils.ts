import {TableEntity} from "../../../components/rating/models"
import {CodingDimension, CodingItem, CodingItemResultByItemId} from "../../../models"

export const itemToTableEntity = (
  item: CodingItem,
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: item.id,
  title: item.title,
  position: item.position,
  score: codingItemResultsByItem[item.id].score,
  maxScore: codingItemResultsByItem[item.id].maximumScore,
  averageScore: codingItemResultsByItem[item.id].averageScore,
  rated: true
})

export const dimensionToTableEntity = (
  dimension: CodingDimension,
  items: CodingItem[],
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: dimension.id,
  title: dimension.title,
  position: dimension.position,
  score: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].score, 0),
  maxScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].maximumScore, 0),
  averageScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].averageScore, 0),
  rated: true
})
