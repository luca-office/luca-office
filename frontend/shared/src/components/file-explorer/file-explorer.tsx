import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import {compact, sortBy, uniq} from "lodash-es"
import * as React from "react"
import {IconName, NodeType} from "../../enums"
import {ButtonConfig, DirectoryNode, FileNode, RootNode, TreeNodeType} from "../../models"
import {useLucaTranslation} from "../../translations"
import {Option} from "../../utils"
import {ContentLoadingIndicator} from ".."
import {TableOfContentsContainer} from "../table-of-content"
import {TreeNode} from "./tree-node"

export interface RenderDirectoryProps {
  node: DirectoryNode
  onClickChevron: (directoryNode: DirectoryNode, event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
  isDirectoryExpanded: boolean
  isDirectorySelected: boolean
  level: number
}

export interface FileExplorerCustomStyles {
  readonly card?: CSSInterpolation
  readonly cardContent?: CSSInterpolation
}

export interface FileExplorerProps {
  readonly tree: RootNode
  readonly isLoading: boolean
  readonly expandedDirectoryIds: UUID[]
  readonly selectedNodeId: Option<UUID>
  readonly onSelectNode?: (node: TreeNodeType) => void
  readonly onExpandDirectory: (directoryNodeId: string) => void
  readonly isCreateDirectoryButtonVisible?: boolean
  readonly onCreateDirectory?: () => void
  readonly renderOnlyDirectories?: boolean
  readonly renderDirectoryNode?: (options: RenderDirectoryProps) => React.ReactNode
  readonly renderCustomTopEntry?: () => React.ReactNode
  readonly renderCustomDirectoryIcon?: (node: DirectoryNode) => IconName
  readonly renderCustomFileSuffix?: (node: FileNode) => string
  readonly customStyles?: FileExplorerCustomStyles
  readonly prependSort?: UUID[]
  readonly appendSort?: UUID[]
  readonly shouldToggleDirectory?: boolean
  readonly countSubDirectoryFiles?: boolean
}

export const FileExplorer: React.FC<FileExplorerProps> = ({
  tree,
  isLoading,
  expandedDirectoryIds,
  onSelectNode,
  onExpandDirectory,
  selectedNodeId,
  isCreateDirectoryButtonVisible,
  onCreateDirectory,
  renderOnlyDirectories,
  renderDirectoryNode,
  renderCustomTopEntry,
  renderCustomDirectoryIcon,
  renderCustomFileSuffix,
  customStyles,
  shouldToggleDirectory,
  prependSort,
  appendSort,
  countSubDirectoryFiles = true
}) => {
  const {t} = useLucaTranslation()

  const sortedTree = React.useMemo<RootNode>(
    () => ({
      ...tree,
      children: getSortedTreeNodes(tree.children, prependSort, appendSort)
    }),
    [tree]
  )

  const renderTree = ({children}: RootNode) =>
    children.map((child, index) => (
      <TreeNode
        key={index}
        node={child}
        level={0}
        renderCustomDirectoryIcon={renderCustomDirectoryIcon}
        renderCustomFileSuffix={renderCustomFileSuffix}
        renderOnlyDirectories={renderOnlyDirectories}
        renderDirectoryNode={renderDirectoryNode}
        selectedNodeId={selectedNodeId}
        expandedDirectoryIds={expandedDirectoryIds}
        onSelect={onSelectNode}
        onExpandDirectory={onExpandDirectory}
        shouldToggleDirectory={shouldToggleDirectory}
        countSubDirectoryFiles={countSubDirectoryFiles}
      />
    ))

  const isTreeEmpty = sortedTree.children.length === 0
  const actionButtonConfig: ButtonConfig = {
    labelKey: "file_explorer__create_new_main_directories",
    icon: IconName.Add,
    onClick: () => onCreateDirectory && onCreateDirectory()
  }

  return (
    <TableOfContentsContainer
      title={t("file_explorer__title")}
      customCardStyles={customStyles?.card}
      customCardContentStyles={customStyles?.cardContent}
      customTocBodyChildrenStyles={isLoading ? styles.tocBodyChildren : undefined}
      headerIcon={IconName.HardDrive}
      addButtonConfig={isCreateDirectoryButtonVisible ? actionButtonConfig : undefined}
      showPlaceHolder={isTreeEmpty}
      placeholderHeadline={t("file_explorer__no_directories_title")}
      placeholderHint={t("file_explorer__no_directories_message")}>
      {isLoading ? (
        <ContentLoadingIndicator customStyles={styles.loadingIndicator} />
      ) : (
        <>
          {renderCustomTopEntry?.()}
          {renderTree(sortedTree)}
        </>
      )}
    </TableOfContentsContainer>
  )
}

export const getSortedTreeNodes = (
  treeNodes: TreeNodeType[],
  prependSort?: UUID[],
  appendSort?: UUID[]
): TreeNodeType[] => {
  const sortedNodes: TreeNodeType[] = sortBy(
    treeNodes.map(treeNode =>
      treeNode.type === NodeType.Directory ? {...treeNode, children: getSortedTreeNodes(treeNode.children)} : treeNode
    ),
    treeNode => treeNode.name.toLowerCase()
  )

  const idsToPrepend = prependSort || []
  const idsToAppend = appendSort || []

  return compact([
    ...idsToPrepend.map(id => sortedNodes.find(n => n.id === id)),
    ...sortedNodes.filter(n => !uniq([...idsToPrepend, ...idsToAppend]).includes(n.id)),
    ...idsToAppend.map(id => sortedNodes.find(n => n.id === id))
  ])
}

const styles = {
  tocBodyChildren: css({
    marginTop: 0,
    width: "100%",
    height: "100%"
  }),
  loadingIndicator: css({
    width: "100%",
    height: "100%",
    boxSizing: "border-box"
  })
}
