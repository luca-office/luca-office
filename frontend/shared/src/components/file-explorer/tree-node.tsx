import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import React, {Fragment} from "react"
import {IconName, NodeType} from "../../enums"
import {DirectoryNode, FileNode, TreeNodeType} from "../../models"
import {
  backgroundColorBright,
  buttonBackgroundColorDisabled,
  fileExplorerSelectedColor,
  Flex,
  fontColor,
  fontColorLight,
  fontFamily,
  spacing,
  spacingLarge,
  spacingMedium,
  spacingSmall,
  spacingTiny,
  textEllipsis,
  TextSize
} from "../../styles"
import {useLucaTranslation} from "../../translations"
import {countFileChildren, iconForBinaryType, Option} from "../../utils"
import {Icon, Text, Tooltip} from "../"
import {RenderDirectoryProps} from "./file-explorer"

export interface TreeNodeProps {
  readonly node: TreeNodeType
  readonly level: number
  readonly selectedNodeId: Option<UUID>
  readonly expandedDirectoryIds: string[]
  readonly onSelect?: (node: TreeNodeType) => void
  readonly onExpandDirectory: (directoryNodeId: string) => void
  readonly renderOnlyDirectories?: boolean
  readonly customNodeStyle?: CSSInterpolation
  readonly renderDirectoryNode?: (options: RenderDirectoryProps) => React.ReactNode
  readonly renderCustomDirectoryIcon?: (node: DirectoryNode) => IconName
  readonly renderCustomFileSuffix?: (node: FileNode) => string
  readonly shouldToggleDirectory?: boolean
  readonly countSubDirectoryFiles?: boolean
}

export const TreeNode: React.FC<TreeNodeProps> = ({
  node,
  level,
  selectedNodeId,
  expandedDirectoryIds,
  onSelect,
  onExpandDirectory,
  renderDirectoryNode,
  renderOnlyDirectories,
  renderCustomDirectoryIcon,
  renderCustomFileSuffix,
  customNodeStyle,
  shouldToggleDirectory = true,
  countSubDirectoryFiles = true
}) => {
  const {t} = useLucaTranslation()
  const isSelected = selectedNodeId.contains(node.id)
  const handleClick = (node: TreeNodeType) => {
    if (onSelect && ((shouldToggleDirectory && node.type === NodeType.Directory) || !isSelected)) {
      onSelect(node)
    }
  }

  const isFileEnabled = node.type === NodeType.File && node.disabled === false

  const renderDirectory = (node: DirectoryNode) => {
    const isDirectoryExpanded = expandedDirectoryIds.includes(node.id)
    const fileCount = countFileChildren(node, countSubDirectoryFiles)
    const hasContent = fileCount > 0 || node.children.length > 0
    const iconColor = hasContent ? fontColor : buttonBackgroundColorDisabled

    const onClickChevron = (directoryNode: DirectoryNode, event: React.MouseEvent<HTMLDivElement>) => {
      if (hasContent) {
        event.stopPropagation()
        onExpandDirectory(directoryNode.id)
      }
    }

    return (
      <Fragment>
        {renderDirectoryNode ? (
          renderDirectoryNode({node, onClickChevron, isDirectoryExpanded, isDirectorySelected: isSelected, level})
        ) : (
          <div
            css={[styles.entry(level, isSelected), customNodeStyle]}
            className="tree-node"
            onClick={() => handleClick(node)}>
            <div css={styles.entryContent(false)}>
              <Tooltip title={!hasContent ? t("files_and_directories__tree__no_content") : ""}>
                <Icon
                  color={iconColor}
                  css={styles.chevronIcon}
                  width={spacingMedium}
                  name={isDirectoryExpanded ? IconName.ChevronFilledDown : IconName.ChevronFilledRight}
                  onClick={event => onClickChevron(node, event)}
                />
              </Tooltip>

              <Icon
                css={{marginRight: spacingSmall}}
                width={spacingMedium}
                name={renderCustomDirectoryIcon !== undefined ? renderCustomDirectoryIcon(node) : IconName.Folder}
              />
              <div className="tree-node-file-count" css={textEllipsis}>{`${node.name} (${fileCount})`}</div>
            </div>
          </div>
        )}

        {isDirectoryExpanded
          ? node.children.map((child, index) => (
              <TreeNode
                key={index}
                node={child}
                level={level + 1}
                customNodeStyle={customNodeStyle}
                renderOnlyDirectories={renderOnlyDirectories}
                renderDirectoryNode={renderDirectoryNode}
                renderCustomFileSuffix={renderCustomFileSuffix}
                selectedNodeId={selectedNodeId}
                expandedDirectoryIds={expandedDirectoryIds}
                onSelect={onSelect}
                onExpandDirectory={onExpandDirectory}
                countSubDirectoryFiles={countSubDirectoryFiles}
              />
            ))
          : null}
      </Fragment>
    )
  }

  const renderFile = (node: FileNode, isVisible: boolean) =>
    isVisible ? (
      <div
        css={[styles.entry(level, isSelected, true), customNodeStyle]}
        onClick={() => isFileEnabled && onSelect?.(node)}>
        <div css={styles.entryContent(!isFileEnabled)}>
          <Icon
            hasSpacing={true}
            color={isFileEnabled ? fontColor : buttonBackgroundColorDisabled}
            customSpacing={spacingSmall}
            name={iconForFileNode(node)}
            width={spacingMedium}
          />
          <div css={styles.fileName}>
            {node.name}
            {renderCustomFileSuffix && (
              <Text customStyles={styles.fileNameSuffix} size={TextSize.Small}>
                {renderCustomFileSuffix(node)}
              </Text>
            )}
          </div>

          {node.isLocked && (
            <Icon name={IconName.LockClosed} color={fontColorLight} customStyles={{marginLeft: spacingSmall}} />
          )}
        </div>
      </div>
    ) : null

  const renderNode = (node: TreeNodeType) => {
    switch (node.type) {
      case NodeType.Directory:
        return renderDirectory(node as DirectoryNode)
      case NodeType.File:
        return renderFile(node as FileNode, !renderOnlyDirectories)
      default:
        return null
    }
  }

  return renderNode(node)
}

const iconForFileNode = (node: FileNode) =>
  node.binaryType !== null ? iconForBinaryType(node.binaryType) : iconForNonBinaryType(node)

const iconForNonBinaryType = (node: FileNode) =>
  node.name.slice(-4) === ".txt" ? IconName.TextEditor : IconName.TableCalculation

const styles = {
  entry: (level: number, isSelected?: boolean, isFile?: boolean) =>
    css({
      cursor: !isSelected ? "pointer" : "default",
      paddingLeft: level * spacingMedium + (isFile ? spacingMedium + spacingTiny : 0),
      backgroundColor: isSelected ? fileExplorerSelectedColor : backgroundColorBright,
      borderRadius: spacingTiny,
      boxSizing: "border-box"
    }),
  entryContent: (isDisabled: boolean) =>
    css(Flex.row, {
      fontFamily: fontFamily,
      fontSize: TextSize.Small,
      lineHeight: spacing(spacingMedium),
      letterSpacing: 0.13,
      height: spacingLarge,
      cursor: isDisabled ? "not-allowed" : "pointer",
      color: isDisabled ? buttonBackgroundColorDisabled : fontColor,
      alignItems: "center"
    }),
  chevronIcon: css({
    marginRight: spacingTiny,
    cursor: "pointer"
  }),
  fileName: css(textEllipsis, Flex.row, {
    flex: "1 1 auto"
  }),
  fileNameSuffix: css(textEllipsis, {
    color: fontColorLight,
    marginLeft: "auto"
  })
}
