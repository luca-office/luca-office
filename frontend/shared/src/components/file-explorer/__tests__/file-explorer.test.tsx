import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {BinaryType, IconName, NodeType} from "../../../enums"
import {FileNode} from "../../../models"
import {Option} from "../../../utils"
import {Button, Card, CardFooter, CardHeader} from "../.."
import {TableOfContentsContainer} from "../../table-of-content/table-of-contents"
import {treeMock} from "../__mocks__/tree.mock"
import {FileExplorer, FileExplorerProps, getSortedTreeNodes} from "../file-explorer"
import {getByText, render, screen} from "@testing-library/react"

const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}

const defaultProps: FileExplorerProps = {
  isLoading: false,
  onSelectNode: jest.fn(),
  expandedDirectoryIds: [],
  onCreateDirectory: jest.fn(),
  onExpandDirectory: jest.fn(),
  selectedNodeId: Option.none(),
  tree: treeMock,
  isCreateDirectoryButtonVisible: false
}

const getComponent = (props?: Partial<FileExplorerProps>) => <FileExplorer {...{...defaultProps, ...props}} />

describe("file-explorer", () => {
  it("renders correctly", () => {
    const fileExplorer = getComponent()
    const tree = create(fileExplorer).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly with custom renderers", () => {
    const fileExplorer = getComponent({
      renderCustomFileSuffix: node => node.id + "test",
      renderCustomDirectoryIcon: node => IconName.Add
    })
    const tree = create(fileExplorer).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correct structure button visible", () => {
    // const fileExplorer = shallow()

    render(getComponent({isCreateDirectoryButtonVisible: true}))
    const createButton = screen.getAllByText("file_explorer__create_new_main_directories")

    expect(createButton.length).toBe(1)
  })

  it("renders correct structure button not visible", () => {
    // const fileExplorer = shallow()

    render(getComponent({isCreateDirectoryButtonVisible: false}))
    const createButton = screen.queryByText("file_explorer__create_new_main_directories")

    expect(createButton).toBeNull()
  })

  it("applies append / prepend logic correctly", () => {
    const nodes = [
      {...fileNodeMock, id: "1"},
      {...fileNodeMock, id: "2"},
      {...fileNodeMock, id: "3"},
      {...fileNodeMock, id: "4"}
    ]

    expect(getSortedTreeNodes(nodes).map(_ => _.id)).toEqual(["1", "2", "3", "4"])
    expect(getSortedTreeNodes(nodes, ["3"]).map(_ => _.id)).toEqual(["3", "1", "2", "4"])
    expect(getSortedTreeNodes(nodes, ["3", "2"]).map(_ => _.id)).toEqual(["3", "2", "1", "4"])
    expect(getSortedTreeNodes(nodes, []).map(_ => _.id)).toEqual(["1", "2", "3", "4"])
    expect(getSortedTreeNodes(nodes, [], ["1"]).map(_ => _.id)).toEqual(["2", "3", "4", "1"])
    expect(getSortedTreeNodes(nodes, ["3"], ["1"]).map(_ => _.id)).toEqual(["3", "2", "4", "1"])

    expect(getSortedTreeNodes(nodes, ["5"]).map(_ => _.id)).toEqual(["1", "2", "3", "4"])
    expect(getSortedTreeNodes(nodes, ["3"], ["3"]).map(_ => _.id)).toEqual(["3", "1", "2", "4", "3"])
  })
})
