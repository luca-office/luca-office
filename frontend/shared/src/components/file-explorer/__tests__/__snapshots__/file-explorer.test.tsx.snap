// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file-explorer renders correctly 1`] = `
<div
  className="css-1tz1azl-Card"
>
  <div
    className="card-header css-15zaw2t-CardHeader"
  >
    <div
      className="css-mv0otu-TableOfContentsContainer"
    >
      <div
        className="css-oivmdv-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1f6p2vj-Heading"
      >
        file_explorer__title
      </div>
    </div>
  </div>
  <div
    className="toc-body css-awggpr-TableOfContentsContainer"
  >
    <div
      className="book-chapters-list css-1n23nny-TableOfContentsContainer"
    >
      <div
        className="css-18xeog2-TableOfContentsContainer"
      >
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-7kj7oz-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Bilder (3)
            </div>
          </div>
        </div>
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-7kj7oz-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Rechnungen (3)
            </div>
          </div>
        </div>
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-1cnfiwv-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              files_and_directories__tree__no_content
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Sonstiges (0)
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-105jaxv-CardFooter"
  />
</div>
`;

exports[`file-explorer renders correctly with custom renderers 1`] = `
<div
  className="css-1tz1azl-Card"
>
  <div
    className="card-header css-15zaw2t-CardHeader"
  >
    <div
      className="css-mv0otu-TableOfContentsContainer"
    >
      <div
        className="css-oivmdv-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1f6p2vj-Heading"
      >
        file_explorer__title
      </div>
    </div>
  </div>
  <div
    className="toc-body css-awggpr-TableOfContentsContainer"
  >
    <div
      className="book-chapters-list css-1n23nny-TableOfContentsContainer"
    >
      <div
        className="css-18xeog2-TableOfContentsContainer"
      >
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-7kj7oz-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Bilder (3)
            </div>
          </div>
        </div>
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-7kj7oz-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Rechnungen (3)
            </div>
          </div>
        </div>
        <div
          className="tree-node css-jvgwtt-TreeNode"
          onClick={[Function]}
        >
          <div
            className="css-ruh0zs-TreeNode"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-1cnfiwv-TreeNode"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
                onClick={[Function]}
              />
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              files_and_directories__tree__no_content
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
            <div
              className="css-1g2afg4-TreeNode"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
            <div
              className="tree-node-file-count css-195yrgn-TreeNode"
            >
              Sonstiges (0)
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-105jaxv-CardFooter"
  />
</div>
`;
